src/engine/ruleset/decoders/windows/windows-security.yml

```diff
@@ -0,0 +1,390 @@
+
+    # TargetUserDomain to user.domain or user.target.domain
+ - check: +ef_not_exists/$user.domain AND +a_contains/~temp.id_target_user_array/$event.code
+   map:
+        - user.domain: $~windows.Event.EventData.TargetDomainName
+ - check: +ef_exists/$user.domain AND +a_contains/~temp.id_target_user_array/$event.code
+   map:
+        - user.target.domain: $~windows.Event.EventData.TargetDomainName
+
+    # Copy MemberName to User and User to Group
+ - check:
+        - ~temp.id_memberName_array: +a_contains/event.code
+   map:
+        - ~temp.split_memberName: +s_to_array/$~windows.Event.EventData.MemberName/,
+        - ~temp.split_memberName.0: '+s_replace/CN=/'
+        - member.name: $~temp.split_memberName.0
+        - related.user: +a_append/$~temp.split_memberName.0
+        - group.id: $~windows.Event.EventData.TargetUserSid
+        - group.id: $~windows.Event.EventData.TargetSid
+        - group.name: $~windows.Event.EventData.TargetUserName
+        - ~windows.Event.EventData.TargetDomainName: '+s_replace/DC=/'
+        - event.goup.domain: $~windows.Event.EventData.TargetDomainName
+        - user.target.group.id: $group.id
+        - user.target.group.name: $group.name
+        - user.target.group.domain: $group.domain
+        - ~temp.split_memberName.3: '+s_replace/DC=/'
+        - user.target.domain: $~temp.split_memberName.3
+
+ # Copy Target User to Computer Object
+ - check: event.code==4741 OR event.code==4742 OR event.code==4743
+   map:
+    - winlog.computerObject.id: $~windows.Event.EventData.TargetSid
+    - winlog.computerObject.name: $~windows.Event.EventData.TargetUserName
+    - winlog.computerObject.domain: $~windows.Event.EventData.TargetDomainName
+
+ - check: event.code==4634 OR event.code==4647 OR event.code==4964
+   map:
+    - winlog.logon.id: $~windows.Event.EventData.TargetLogonId
+
+ # Copy Subject User from Event Data
+ - check:
+     - ~temp.id_copy_subject_user_array: +a_contains/$event.code
+   map:
+     - user.id: $~windows.Event.EventData.SubjectUserSid
+     - user.name: $~windows.Event.EventData.SubjectUserName
+     - related.user: +a_append/$~windows.Event.EventData.SubjectUserName
+     - user.domain: $~windows.Event.EventData.SubjectDomainName
+
+ # Copy Target User to Target
+ - check:
+    - ~temp.id_copy_target_user_array: +a_contains/$event.code
+   map:
+     - user.target.id: $~windows.Event.EventData.TargetSid
+     - ~temp.name_part: +s_to_array/$~windows.Event.EventData.TargetUserName/@
+     - user.target.name: $~temp.name_part.0
+     - related.user: +a_append/$event.user.target.name
+     - user.target.domain: $~windows.Event.EventData.TargetDomainName
+
+# Copy Target User to Effective
+ - check: event.code==4648 OR event.code==4688
+   map:
+     - user.effective.id: $~windows.Event.EventData.TargetUserSid
+
+
+ - check: +s_ne/$~windows.Event.EventData.TargetUserName/- AND  (event.code==4648 OR event.code==4688)
```

Comment:
I'm not a 100% sure that the parenthesis do work wth operators, can we have a set of example events in order to test this?